File: bin/my_scripts/utils/format_merge_lines.scala
// Scala, being a functional and object-oriented programming language that runs
// on the Java Virtual Machine, is well-suited for tasks involving file
// operations and string manipulation.

import java.nio.file.{Files, Paths, Path}
import scala.collection.JavaConverters._

object FileProcessor extends App {
  def checkFirstLine(line: String): Boolean = {
    val tempLine = line.trim
    (tempLine.isEmpty || tempLine.startsWith("/") || tempLine.startsWith("*")) && !tempLine.startsWith("#")
  }

  def checkForComment(line: String): Boolean = {
    line.trim.startsWith("/") || line.trim.startsWith("*")
  }

  def checkForContent(line: String): Boolean = {
    line.trim.nonEmpty && !checkForComment(line) && !line.trim.startsWith("#")
  }

  def checkForParen(line: String): Boolean = {
    line.trim.contains("(")
  }

  def checkForEndParen(line: String): Boolean = {
    line.trim.endsWith(")")
  }

  def checkForFuncStart(line: String): Boolean = {
    line.replace("\n", "").startsWith("{")
  }

  def processFile(path: Path): Int = {
    val lines = Files.readAllLines(path).asScala
    val newLines = scala.collection.mutable.ArrayBuffer[String]()
    var changeCounter = 0
    var i = 0

    while (i < lines.length) {
      if (i < lines.length - 3 && checkFirstLine(lines(i)) && checkForContent(lines(i + 1)) && checkForEndParen(lines(i + 2)) && checkForFuncStart(lines(i + 3))) {
        newLines += lines(i)
        newLines += lines(i + 1).trim + " " + lines(i + 2).trim
        newLines += lines(i + 3)
        i += 4
        changeCounter += 1
      } else {
        newLines += lines(i)
        i += 1
      }
    }

    Files.write(path, newLines.asJava)
    changeCounter
  }

  val dirPath = "./" // Current directory
  var totalChangeCount = 0

  Files.newDirectoryStream(Paths.get(dirPath), "*.c").asScala.foreach { path =>
    println(s"Processing file: ${path.getFileName}")
    val fileChangeCount = processFile(path)
    println(s"Done! Changes done: $fileChangeCount")
    totalChangeCount += fileChangeCount
  }

  println(s"Done processing files! Total changes done: $totalChangeCount")
}

File: bin/my_scripts/utils/format_merge_lines.php
<?php

function checkFirstLine($line) {
    $tempLine = trim($line);
    return ($tempLine === '' || strpos($tempLine, '/') === 0 || strpos($tempLine, '*') === 0) && strpos($tempLine, '#') !== 0;
}

function checkForComment($line) {
    return strpos(trim($line), '/') === 0 || strpos(trim($line), '*') === 0;
}

function checkForContent($line) {
    return trim($line) !== '' && !checkForComment($line) && strpos(trim($line), '#') !== 0;
}

function checkForParen($line) {
    return strpos(trim($line), '(') !== false;
}

function checkForEndParen($line) {
    return substr(trim($line), -1) === ')';
}

function checkForFuncStart($line) {
    return strpos(str_replace("\n", '', $line), '{') === 0;
}

function processFile($filename) {
    $lines = file($filename, FILE_IGNORE_NEW_LINES);
    $newLines = [];
    $changeCounter = 0;

    $i = 0;
    while ($i < count($lines)) {
        if ($i < count($lines) - 3 && checkFirstLine($lines[$i]) && checkForContent($lines[$i + 1]) && checkForEndParen($lines[$i + 2]) && checkForFuncStart($lines[$i + 3])) {
            array_push($newLines, $lines[$i]);
            array_push($newLines, trim($lines[$i + 1]) . ' ' . trim($lines[$i + 2]));
            array_push($newLines, $lines[$i + 3]);
            $i += 4;
            $changeCounter++;
        } else {
            array_push($newLines, $lines[$i]);
            $i++;
        }
    }

    file_put_contents($filename, implode("\n", $newLines));
    return $changeCounter;
}

$dirPath = './'; // Current directory
$totalChangeCount = 0;

foreach (new DirectoryIterator($dirPath) as $fileInfo) {
    if ($fileInfo->isDot()) continue;
    if ($fileInfo->getExtension() === 'c') {
        echo "Processing file: " . $fileInfo->getFilename() . "\n";
        $fileChangeCount = processFile($fileInfo->getPathname());
        echo "Done! Changes done: " . $fileChangeCount . "\n";
        $totalChangeCount += $fileChangeCount;
    }
}

echo "Done processing files! Total changes done: " . $totalChangeCount . "\n";
?>
